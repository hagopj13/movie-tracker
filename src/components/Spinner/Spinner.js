import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: theme.spacing(3, 0),
  },
}));

const Spinner = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <CircularProgress color="primary" />
    </div>
  );
};

export default Spinner;
