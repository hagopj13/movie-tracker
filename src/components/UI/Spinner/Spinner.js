import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: theme.spacing(3, 0),
  },
}));

export default function CircularIndeterminate() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <CircularProgress color="primary" />
    </div>
  );
}
