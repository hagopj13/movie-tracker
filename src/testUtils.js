import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { createMount } from '@material-ui/core/test-utils';

import { initialState } from 'store/rootReducer';

const mount = createMount();
const mockStore = configureMockStore()(initialState);

export const mountWithStore = (children, store = mockStore) =>
  mount(<Provider store={store}>{children}</Provider>);
