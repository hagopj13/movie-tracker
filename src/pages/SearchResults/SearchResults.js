// @flow
import React, { useLayoutEffect } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';

import Spinner from 'components/Spinner/Spinner';
import searchActions from 'store/search/search.actions';
import searchSelectors from 'store/search/search.selectors';
import SearchActionTypes from 'store/search/search.types';
import loadingSelectors from 'store/api/loading/loading.selectors';
import ConfigActionTypes from 'store/config/config.types';

import SearchResultsOverview from './Overview/SearchResultsOverview';
import SearchResultsList from './List/SearchResultsList';

type Props = {
  isLoading: boolean,
  selectedSearchQuery: string,
  onFetchMovies: () => void,
};

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
  },
  container: {
    padding: theme.spacing(5, 5, 2),
    display: 'flex',
    flexDirection: 'column',
  },
  overview: {
    marginBottom: theme.spacing(3),
  },
}));

export const SearchResults = (props: Props) => {
  const { isLoading, selectedSearchQuery, onFetchMovies } = props;

  const classes = useStyles();

  useLayoutEffect(() => {
    if (selectedSearchQuery) {
      onFetchMovies();
    }
  }, [onFetchMovies, selectedSearchQuery]);

  if (isLoading) {
    return <Spinner />;
  }

  if (!selectedSearchQuery) {
    return null;
  }

  return (
    <div className={classes.root}>
      <Container className={classes.container}>
        <div className={classes.overview}>
          <SearchResultsOverview />
        </div>
        <div>
          <SearchResultsList />
        </div>
      </Container>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  isLoading: loadingSelectors.createIsLoadingSelector([
    ConfigActionTypes.FETCH_IMAGES_CONFIG,
    SearchActionTypes.FETCH_MOVIES,
  ]),
  selectedSearchQuery: searchSelectors.selectQuery,
});

const mapDispatchToProps = {
  onFetchMovies: searchActions.fetchMovies,
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchResults);
