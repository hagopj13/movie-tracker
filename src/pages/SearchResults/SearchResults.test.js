import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { mountWithStore } from 'testUtils';
import Spinner from 'components/Spinner/Spinner';

import { SearchResults } from './SearchResults';

describe('SearchResults page', () => {
  let shallow;
  let wrapper;

  const mockFetchMovies = jest.fn();

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render the SearchResults page correctly', () => {
    const mockProps = {
      isLoading: false,
      selectedSearchQuery: 'some search query',
      onFetchMovies: mockFetchMovies,
    };
    wrapper = shallow(<SearchResults {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a Spinner when it is loading', () => {
    const mockProps = {
      isLoading: true,
      selectedSearchQuery: 'some search query',
      onFetchMovies: mockFetchMovies,
    };
    wrapper = shallow(<SearchResults {...mockProps} />);
    expect(wrapper.exists(Spinner)).toBe(true);
  });

  it('should not render anything when there is no selected search query', () => {
    const mockProps = {
      isLoading: false,
      selectedSearchQuery: null,
      onFetchMovies: mockFetchMovies,
    };
    wrapper = shallow(<SearchResults {...mockProps} />);
    expect(wrapper.isEmptyRender()).toBe(true);
    expect(mockFetchMovies).not.toHaveBeenCalled();
  });

  it('should fetch movies after mounting', () => {
    const mockProps = {
      isLoading: false,
      selectedSearchQuery: 'some search query',
      onFetchMovies: mockFetchMovies,
    };
    wrapper = mountWithStore(<SearchResults {...mockProps} />);
    expect(mockFetchMovies).toHaveBeenCalledTimes(1);
  });
});
