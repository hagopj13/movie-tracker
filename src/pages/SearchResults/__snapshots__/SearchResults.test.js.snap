// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults page should render the SearchResults page correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <div
      className="makeStyles-overview-3"
    >
      <Connect(SearchResultsOverview) />
    </div>
    <div>
      <Connect(SearchResultsList) />
    </div>
  </WithStyles(ForwardRef(Container))>
</div>
`;
